taca/illumina/HiSeqX_Runs.py

```diff
@@ -289,13 +304,22 @@ def _generate_samplesheet_subset(ssparser, lanes):
     output+=",".join(datafields)
     output+=os.linesep
     for line in ssparser.data:
-        if line['Lane'] in lanes:
-            line_ar=[]
-            for field in datafields:
-                if field == "index" and "NOINDEX" in line[field].upper():
-                    line[field] = ""
-                line_ar.append(line[field])
-            output+=",".join(line_ar)
-            output+=os.linesep
+        sample_name = line.get('Sample_Name') or line.get('SampleName')
+        if line['Lane'] in lanes and is_10X:
+            if sample_name in tenX_samples[line['Lane']]:
+                line_ar=[]
+                for field in datafields:
+                    line_ar.append(line[field])
+                output+=",".join(line_ar)
+                output+=os.linesep
+        elif line['Lane'] in lanes and not is_10X:
+            if tenX_samples.get(line['Lane'],'') == '' or sample_name not in tenX_samples.get(line['Lane']):
```

Comment:
`tenX_samples.get(line['Lane'],'') == ''` is the same thing as `line['Lane'] not in tenX_samples` right?